objects/src/accounts/delta/vault.rs

```diff
@@ -37,6 +41,99 @@ impl AccountVaultDelta {
         }
     }
 
+    /// Merges another delta into this one, overwriting any existing values.
+    ///
+    /// Inputs and the result are validated as part of the merge.
+    pub fn merge(self, other: Self) -> Result<Self, AccountDeltaError> {
```

Comment:
Looking at how painful this is, I very much agree that we should refactor the delta to look something like this:
```rust
pub struct AccountVaultDelta {
    fungible: BTreeMap<AccountId, i64>,
    non_fungible: BTreeMap<NonFungibleAsset, bool>,
}
```
But since this is already working, we can probably leave the refactoring to #798.

One potential caveat that the ordering of assets may become important in the future - but we can cross that bridge when we get there.